From bfc6eeb678d3a16ad087aa6b5b3451fa71c58474 Mon Sep 17 00:00:00 2001
From: Farid CHABANE <farid-externe.chabane@edf.fr>
Date: Wed, 12 Nov 2025 20:40:52 +0100
Subject: [PATCH 2/3] cython encoding bug

There's an upstream bug in cython which will result in an incorrect
encoding detection if the c_string_encoding keyword appears in the first
two lines of a file: cython/cython#2632

So the "simple" fix is to add 2 extra lines before the magic headers of
the only files which is incorrectly detected.

This patch is almost identical to the one used when packaging with
Slurm 24.05 with one less file affected this time.
---
 pyslurm/slurm/__init__.pxd | 2 ++
 pyslurm/utils/__init__.pxd | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/pyslurm/slurm/__init__.pxd b/pyslurm/slurm/__init__.pxd
index ff15178..8554785 100644
--- a/pyslurm/slurm/__init__.pxd
+++ b/pyslurm/slurm/__init__.pxd
@@ -1,3 +1,5 @@
+
+
 # cython: c_string_type=unicode, c_string_encoding=default
 # cython: language_level=3
 
diff --git a/pyslurm/utils/__init__.pxd b/pyslurm/utils/__init__.pxd
index 7a22bfa..231a512 100644
--- a/pyslurm/utils/__init__.pxd
+++ b/pyslurm/utils/__init__.pxd
@@ -1,2 +1,4 @@
+
+
 # cython: c_string_type=unicode, c_string_encoding=default
 # cython: language_level=3
-- 
2.30.2

